aten/src/ATen/native/mkldnn/MKLDNNHandle.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+
+#if AT_MKLDNN_ENABLED()
+
+#include <ideep.hpp>
+
+#include "c10/util/intrusive_ptr.h"
+
+namespace at { namespace native {
+
+struct CAFFE2_API MKLDNNHandle : c10::intrusive_ptr_target {
```

Comment:
This MKLDNNHandle doesn't seem necessary to me. Can you directly put `ideep::tensor` to the `TensorImpl::opaque_handle_`?

Reply:
> This MKLDNNHandle doesn't seem necessary to me. Can you directly put `ideep::tensor` to the `TensorImpl::opaque_handle_`?

@bddppq `MKLDNNHandle` was created to inherit `c10::intrusive_ptr_target`. Yes, it is possible to make `ideep::tensor` inherit `c10::intrusive_ptr_target` too but since IDEEP is currently on a third-party repo, it is not good to make it depend on c10. We are working on integrate IDEEP code into PyTorch. How about we refactor the code afterwards?

Reply:
Ah I see. In this case naming it to OpaqueHandle (template class) would be even better (so it can be reused by other opaque tensors as they will face the same issue).

Reply:
@bddppq Hmm, I named it `MKLDNNHandle` rather than `OpaqueHandle` since it wraps MKL-DNN specific stuff like `ideep::tensor`. I don't find the benefit to introduce another level of abstraction `OpaqueHandle` here. A simpler way for other opaque tensors to do is to similarly inherit `c10::intrusive_ptr_target` and wrap their own stuff. It is more straight-forward IMHO.

Reply:
Make it a template class? I tbh don't hope to see slightly different `XYZHandle` classes added later.

Reply:
@bddppq OK. What's the good place to add the file? Any suggestion?

Reply:
@bddppq How about c10/core/OpaqueHandle.h?

Reply:
@bddppq Or simply put into TensorImpl.h...

Reply:
Let's put it to OpaqueHandle.h